lib/Canonicalize/SwitchToStatements.cpp

```diff
@@ -997,9 +996,8 @@ bool VariableBuilder::serializeToLocalVariable(Instruction *I) {
   // variables, because their initialization (which is forcibly out-of-line)
   // would assign them and fail to compile.
   // For this reason if at this point we're trying to declare a constant
-  // local variable, we're forced to throw away the const-qualifier.
-  if (VariableType.isConst())
-    VariableType = getNonConst(VariableType);
+  // local variable, we're forced to throw away the constness information.
+  const_cast<model::Type &>(*VariableType).stripConstness();
```

Comment:
Yeah, but the alternative is to rework a whole lot of stuff to use a non-const model. For a temporary thing at that.

Reply:
The old code didn't have it?
Can you elaborate why this is now needed?

Reply:
Actually, on the fresh look, maybe the old way of "discarding" typedefs instead of "fixing" them is better. I guess I have been overlooking the fact that the relevant header is likely to be produced from the original model instead, which would lead to problems. I wonder why it never showed up in the cases I used to test this thing on 🤔.

Reply:
Done.